cmd/root.go

```diff
@@ -241,7 +241,18 @@ func (c *rootCommand) persistentPreRunE(cmd *cobra.Command, args []string) error
 		c.loggerIsRemote = true
 	}
 
-	stdlog.SetOutput(c.globalState.logger.Writer())
+	// Sometimes the Go runtime uses the standard log output to
+	// log some messages directly.
+	// It does when an invalid char is found in a Cookie.
+	// Check for details https://github.com/grafana/k6/issues/711#issue-341414887
+	stdlog.SetOutput(func() io.Writer {
+		w := c.globalState.logger.Writer()
+		go func() {
+			<-c.globalState.ctx.Done()
+			_ = w.Close()
+		}()
+		return w
```

Comment:
I am not sure this is sufficient and won't cause problems :thinking: I can think of 2 potential problems. :thinking: 

First, if something somehow logged to the stdlib logger after this writer was closed, I imagine this will panic, which is not ideal. It should not happen, but if it happens, it's better to not panic.

Second, I am not sure this internal goroutine will never be detected by the goroutine detector :thinking:  After all, it waits for the `globalState.ctx` to be done and closes something, but nothing actually waits for the goroutine to be finished before `execute()` is done.

Reply:
Off the top of my head, better approach would be a `io.Writer` that either discards incoming logs or falls back to the `fallbackLogger` once `globalState.ctx` is done, this way it won't panic. And something similar to `waitRemoteLogger()` that waits for this goroutine to end before `execute()` finishes, to not leave it potentially dangling.

Reply:
>something similar to waitRemoteLogger() that waits for this goroutine to end before execute() finishes, to not leave it potentially dangling.

@na-- I refactored a bit as you asked. Check the latest version, please.

>First, if something somehow logged to the stdlib logger after this writer was closed, I imagine this will panic, which is not ideal. It should not happen, but if it happens, it's better to not panic.

The `io.Write` API defines error as returned type, so I expect it will not generate any panic in that case, and log is a stdlib package, so I expect some consistency respecting the design.

>Off the top of my head, better approach would be a io.Writer that either discards incoming logs or falls back to the fallbackLogger once globalState.ctx is done

Does it make sense to have this dedicated to this part of the code? Why does not add a global component that automatically handles it?